Build problem for mksurfdata tools testers

### Brief summary of bug

[Give a one or two sentence summary. This could be the same as the issue title if you feel that is a sufficient summary.]

### General bug information

**CTSM version you are using:** ctsm5.1.dev066

**Does this bug cause significantly incorrect results in the model's science?** No

**Configurations affected:** Some of the mksurfdata unit testers
### Details of bug

Looks like some options in the tests aren't updated to use the new interfaces for certain subroutines that had an interface change.

@slevisconsulting I think this was your change you made in ctsm5.1.dev040. And is likely pretty straightforward to fix.

### Important details of your setup / configuration so we can reproduce the bug

This is on cheyenne with intel compiler going to the directory

tools/mksurfdata_map/unit_testers

and running

gmake

### Important output or errors that show the problem

Heres the build problem I see...

```
l/include -I/usr/local/lib   test_mkindexmapMod.F90
test_mkindexmapMod.F90(77): error #6631: A non-optional actual argument must be present when invoking a procedure with an explicit interface.   [MASK_SRC]
      call get_dominant_indices(gridmap, src_array, dst_array, minval, maxval, nodata)
-----------^
test_mkindexmapMod.F90(86): error #6631: A non-optional actual argument must be present when invoking a procedure with an explicit interface.   [MASK_SRC]
      call get_dominant_indices(gridmap, src_array, dst_array, minval, maxval, nodata)
-----------^
test_mkindexmapMod.F90(95): error #6631: A non-optional actual argument must be present when invoking a procedure with an explicit interface.   [MASK_SRC]
      call get_dominant_indices(gridmap, src_array, dst_array, minval, maxval, nodata)
-----------^
test_mkindexmapMod.F90(104): error #6631: A non-optional actual argument must be present when invoking a procedure with an explicit interface.   [MASK_SRC]
      call get_dominant_indices(gridmap, src_array, dst_array, minval, maxval, nodata)
-----------^
test_mkindexmapMod.F90(113): error #6631: A non-optional actual argument must be present when invoking a procedure with an explicit interface.   [MASK_SRC]
      call get_dominant_indices(gridmap, src_array, dst_array, minval, maxval, nodata)
-----------^
test_mkindexmapMod.F90(122): error #6631: A non-optional actual argument must be present when invoking a procedure with an explicit interface.   [MASK_SRC]
      call get_dominant_indices(gridmap, src_array, dst_array, minval, maxval, nodata)
-----------^
test_mkindexmapMod.F90(134): error #6631: A non-optional actual argument must be present when invoking a procedure with an explicit interface.   [MASK_SRC]
      call get_dominant_indices(gridmap, src_array, dst_array, minval, maxval, nodata, filter=filter)
-----------^
test_mkindexmapMod.F90(146): error #6631: A non-optional actual argument must be present when invoking a procedure with an explicit interface.   [MASK_SRC]
      call get_dominant_indices(gridmap, src_array, dst_array, minval, maxval, nodata, filter=filter)
-----------^
test_mkindexmapMod.F90(159): error #6631: A non-optional actual argument must be present when invoking a procedure with an explicit interface.   [MASK_SRC]
      call get_dominant_indices(gridmap, src_array, dst_array, minval, maxval, nodata)
-----------^
test_mkindexmapMod.F90(194): error #6631: A non-optional actual argument must be present when invoking a procedure with an explicit interface.   [MASK_SRC]
      call get_dominant_indices(gridmap, src_array, dst_array, minval, maxval, nodata)
-----------^
test_mkindexmapMod.F90(269): error #6632: Keyword arguments are invalid without an explicit interface.   [NODATA]
      call filter_same(gridmap, filter, src_array, dst_array, nodata=nodata)
--------------------------------------------------------------^
test_mkindexmapMod.F90(288): error #6632: Keyword arguments are invalid without an explicit interface.   [NODATA]
      call filter_same(gridmap, filter, src_array, dst_array, nodata=nodata)
--------------------------------------------------------------^
test_mkindexmapMod.F90(298): error #6632: Keyword arguments are invalid without an explicit interface.   [NODATA]
      call filter_same(gridmap, filter, src_array, dst_array, nodata=nodata)
--------------------------------------------------------------^
compilation aborted for test_mkindexmapMod.F90 (code 1)
../src/Makefile.common:348: recipe for target 'test_mkindexmapMod.o' failed
gmake: *** [test_mkindexmapMod.o] Error 1

```

--- a/tools/mksurfdata_map/unit_testers/test_mkgridmapMod.F90
+++ b/tools/mksurfdata_map/unit_testers/test_mkgridmapMod.F90
@@ -9,16 +9,15 @@ module test_mkgridmapMod
   private
 
   public :: test_gridmap_areastddev
-  public :: test_gridmap_areaave_default
+  public :: test_gridmap_areaave_no_srcmask
   public :: test_gridmap_areaave_srcmask
-  public :: test_gridmap_areaave_srcmask2
   
   character(len=*), parameter :: modname = 'test_mkgridmapMod'
 
 contains
 
   !------------------------------------------------------------------------------
-  subroutine test_gridmap_areaave_default
+  subroutine test_gridmap_areaave_no_srcmask
 
     implicit none
 
@@ -32,7 +31,7 @@ contains
     real(r8), parameter :: nodata = -1._r8
     real(r8), parameter :: eps = 1.e-13_r8
 
-    character(len=*), parameter :: subname = 'test_gridmap_areaave_default'
+    character(len=*), parameter :: subname = 'test_gridmap_areaave_no_srcmask'
 
     ! Note about the gridmaps for the tests here:
     ! For most tests here, the test arrays are: (1) simple case, (2) the main case to
@@ -61,7 +60,7 @@ contains
     testname = 'no overlap'
     src_array = (/0.1_r8,0.2_r8,0.3_r8,0.4_r8/)
     dst_array_t = (/0.125_r8, nodata, 0.375_r8/)
-    call gridmap_areaave(gridmap, src_array, dst_array, nodata)
+    call gridmap_areaave_no_srcmask(gridmap, src_array, dst_array, nodata)
     call test_close(dst_array, dst_array_t, eps, modname//' -- '//subname//' -- '//trim(testname))
     deallocate(gridmap%src_indx, gridmap%dst_indx, gridmap%wovr, gridmap%frac_dst)
     deallocate(src_array, dst_array, dst_array_t)
@@ -87,7 +86,7 @@ contains
     testname = 'single overlap'
     src_array = (/0.1_r8,0.2_r8,0.5_r8,0.3_r8,0.4_r8/)
     dst_array_t = (/0.125_r8, 0.5_r8, 0.375_r8/)
-    call gridmap_areaave(gridmap, src_array, dst_array, nodata)
+    call gridmap_areaave_no_srcmask(gridmap, src_array, dst_array, nodata)
     call test_close(dst_array, dst_array_t, eps, modname//' -- '//subname//' -- '//trim(testname))
 
     ! Now change the overlap point to have weight=0
@@ -95,7 +94,7 @@ contains
     gridmap%wovr(3) = 0.0_r8
     gridmap%frac_dst(2) = 0.0_r8
     dst_array_t(2) = nodata
-    call gridmap_areaave(gridmap, src_array, dst_array, nodata)
+    call gridmap_areaave_no_srcmask(gridmap, src_array, dst_array, nodata)
     call test_close(dst_array, dst_array_t, eps, modname//' -- '//subname//' -- '//trim(testname))
 
     deallocate(gridmap%src_indx, gridmap%dst_indx, gridmap%wovr, gridmap%frac_dst)
@@ -130,13 +129,13 @@ contains
     testname='multiple overlaps, all the same value'
     src_array = (/0.1_r8, 0.5_r8, 0.5_r8, 0.5_r8, 0.5_r8, 0.5_r8, 0.5_r8, 0.5_r8, 0.6_r8/)
     dst_array_t = (/0.2_r8, 0.5_r8, 0.575_r8/)
-    call gridmap_areaave(gridmap, src_array, dst_array, nodata)
+    call gridmap_areaave_no_srcmask(gridmap, src_array, dst_array, nodata)
     call test_close(dst_array, dst_array_t, eps, modname//' -- '//subname//' -- '//trim(testname))
 
     testname='multiple overlaps, different values'
     src_array = (/0.1_r8, 0.2_r8, 0.3_r8, 0.7_r8, 0.5_r8, 1.5_r8, 0.5_r8, 1.7_r8, 1.8_r8/)
     dst_array_t = (/0.125_r8, 0.875_r8, 1.775_r8/)
-    call gridmap_areaave(gridmap, src_array, dst_array, nodata)
+    call gridmap_areaave_no_srcmask(gridmap, src_array, dst_array, nodata)
     call test_close(dst_array, dst_array_t, eps, modname//' -- '//subname//' -- '//trim(testname))
 
     ! dividing the weights by 2 shouldn't affect the mean
@@ -145,7 +144,7 @@ contains
     gridmap%frac_dst(:) = gridmap%frac_dst(:) / 2.0_r8
     src_array = (/0.1_r8, 0.2_r8, 0.3_r8, 0.7_r8, 0.5_r8, 1.5_r8, 0.5_r8, 1.7_r8, 1.8_r8/)
     dst_array_t = (/0.125_r8, 0.875_r8, 1.775_r8/)
-    call gridmap_areaave(gridmap, src_array, dst_array, nodata)
+    call gridmap_areaave_no_srcmask(gridmap, src_array, dst_array, nodata)
     call test_close(dst_array, dst_array_t, eps, modname//' -- '//subname//' -- '//trim(testname))
     ! restore wovr & frac_dst
     gridmap%wovr(:) = gridmap%wovr(:) * 2.0_r8
@@ -157,7 +156,7 @@ contains
     gridmap%frac_dst(:) = gridmap%frac_dst(:) * 2.0_r8
     src_array = (/0.1_r8, 0.2_r8, 0.3_r8, 0.7_r8, 0.5_r8, 1.5_r8, 0.5_r8, 1.7_r8, 1.8_r8/)
     dst_array_t = (/0.125_r8, 0.875_r8, 1.775_r8/)
-    call gridmap_areaave(gridmap, src_array, dst_array, nodata)
+    call gridmap_areaave_no_srcmask(gridmap, src_array, dst_array, nodata)
     call test_close(dst_array, dst_array_t, eps, modname//' -- '//subname//' -- '//trim(testname))
     ! restore wovr & frac_dst
     gridmap%wovr(:) = gridmap%wovr(:) / 2.0_r8
@@ -165,7 +164,7 @@ contains
 
     deallocate(src_array, dst_array, dst_array_t)
 
-  end subroutine test_gridmap_areaave_default
+  end subroutine test_gridmap_areaave_no_srcmask
 
   !------------------------------------------------------------------------------
   subroutine test_gridmap_areaave_srcmask
@@ -176,7 +175,7 @@ contains
     character(len=128) :: testname
 
     real(r8), allocatable :: src_array(:)
-    real(r8), allocatable :: mask_src(:)
+    integer , allocatable :: mask_src(:)
     real(r8), allocatable :: dst_array(:)
     real(r8), allocatable :: dst_array_t(:)
 
@@ -212,9 +211,9 @@ contains
          dst_array_t(gridmap%nb))
     testname = 'no overlap'
     src_array = (/0.1_r8,0.2_r8,0.3_r8,0.4_r8/)
-    mask_src(:) = 1.0_r8
+    mask_src(:) = 1
     dst_array_t = (/0.125_r8, nodata, 0.375_r8/)
-    call gridmap_areaave(gridmap, src_array, dst_array, nodata, mask_src)
+    call gridmap_areaave_srcmask(gridmap, src_array, dst_array, nodata, mask_src=mask_src, frac_dst=gridmap%frac_dst)
     call test_close(dst_array, dst_array_t, eps, modname//' -- '//subname//' -- '//trim(testname))
     deallocate(gridmap%src_indx, gridmap%dst_indx, gridmap%wovr, gridmap%frac_dst)
     deallocate(src_array, mask_src, dst_array, dst_array_t)
@@ -242,14 +241,14 @@ contains
     src_array = (/0.1_r8,0.2_r8,0.5_r8,0.3_r8,0.4_r8/)
     mask_src(:) = 1.0_r8
     dst_array_t = (/0.125_r8, 0.5_r8, 0.375_r8/)
-    call gridmap_areaave(gridmap, src_array, dst_array, nodata, mask_src)
+    call gridmap_areaave_srcmask(gridmap, src_array, dst_array, nodata, mask_src=mask_src, frac_dst=gridmap%frac_dst)
     call test_close(dst_array, dst_array_t, eps, modname//' -- '//subname//' -- '//trim(testname))
 
     ! Now change the overlap point to have src_mask=0
     testname = 'single overlap with 0 src_mask'
     mask_src(3) = 0.0_r8
     dst_array_t(2) = nodata
-    call gridmap_areaave(gridmap, src_array, dst_array, nodata, mask_src)
+    call gridmap_areaave_srcmask(gridmap, src_array, dst_array, nodata, mask_src=mask_src, frac_dst=gridmap%frac_dst)
     call test_close(dst_array, dst_array_t, eps, modname//' -- '//subname//' -- '//trim(testname))
 
     deallocate(gridmap%src_indx, gridmap%dst_indx, gridmap%wovr, gridmap%frac_dst)
@@ -286,21 +285,21 @@ contains
     src_array = (/0.1_r8, 0.5_r8, 0.5_r8, 0.5_r8, 0.5_r8, 0.5_r8, 0.5_r8, 0.5_r8, 0.6_r8/)
     mask_src(:) = 1.0_r8
     dst_array_t = (/0.2_r8, 0.5_r8, 0.575_r8/)
-    call gridmap_areaave(gridmap, src_array, dst_array, nodata, mask_src)
+    call gridmap_areaave_srcmask(gridmap, src_array, dst_array, nodata, mask_src=mask_src, frac_dst=gridmap%frac_dst)
     call test_close(dst_array, dst_array_t, eps, modname//' -- '//subname//' -- '//trim(testname))
 
     testname='multiple overlaps, different values'
     src_array = (/0.1_r8, 0.2_r8, 0.3_r8, 0.7_r8, 0.5_r8, 1.5_r8, 0.5_r8, 1.7_r8, 1.8_r8/)
     mask_src(:) = 1.0_r8
     dst_array_t = (/0.125_r8, 0.875_r8, 1.775_r8/)
-    call gridmap_areaave(gridmap, src_array, dst_array, nodata, mask_src)
+    call gridmap_areaave_srcmask(gridmap, src_array, dst_array, nodata, mask_src=mask_src, frac_dst=gridmap%frac_dst)
     call test_close(dst_array, dst_array_t, eps, modname//' -- '//subname//' -- '//trim(testname))
 
     testname='multiple overlaps, different values, srcmask'
     src_array = (/0.1_r8, 0.2_r8, 0.3_r8, 0.7_r8, 0.5_r8, 1.5_r8, 0.5_r8, 1.7_r8, 1.8_r8/)
     mask_src(:) = (/1.0_r8, 1.0_r8, 0.0_r8, 0.5_r8, 1.0_r8, 0.5_r8, 0.0_r8, 1.0_r8, 1.0_r8/)
     dst_array_t = (/0.125_r8, 0.923076923076923_r8, 1.775_r8/)
-    call gridmap_areaave(gridmap, src_array, dst_array, nodata, mask_src)
+    call gridmap_areaave_srcmask(gridmap, src_array, dst_array, nodata, mask_src=mask_src, frac_dst=gridmap%frac_dst)
     call test_close(dst_array, dst_array_t, eps, modname//' -- '//subname//' -- '//trim(testname))
 
     ! dividing the weights by 2 and dividing mask_src by a constant shouldn't affect the mean
@@ -310,7 +309,7 @@ contains
     src_array = (/0.1_r8, 0.2_r8, 0.3_r8, 0.7_r8, 0.5_r8, 1.5_r8, 0.5_r8, 1.7_r8, 1.8_r8/)
     mask_src(:) = 0.25_r8
     dst_array_t = (/0.125_r8, 0.875_r8, 1.775_r8/)
-    call gridmap_areaave(gridmap, src_array, dst_array, nodata, mask_src)
+    call gridmap_areaave_srcmask(gridmap, src_array, dst_array, nodata, mask_src=mask_src, frac_dst=gridmap%frac_dst)
     call test_close(dst_array, dst_array_t, eps, modname//' -- '//subname//' -- '//trim(testname))
     ! restore wovr & frac_dst
     gridmap%wovr(:) = gridmap%wovr(:) * 2.0_r8
@@ -323,7 +322,7 @@ contains
     src_array = (/0.1_r8, 0.2_r8, 0.3_r8, 0.7_r8, 0.5_r8, 1.5_r8, 0.5_r8, 1.7_r8, 1.8_r8/)
     mask_src(:) = 0.25_r8
     dst_array_t = (/0.125_r8, 0.875_r8, 1.775_r8/)
-    call gridmap_areaave(gridmap, src_array, dst_array, nodata, mask_src)
+    call gridmap_areaave_srcmask(gridmap, src_array, dst_array, nodata, mask_src=mask_src, frac_dst=gridmap%frac_dst)
     call test_close(dst_array, dst_array_t, eps, modname//' -- '//subname//' -- '//trim(testname))
     ! restore wovr & frac_dst
     gridmap%wovr(:) = gridmap%wovr(:) / 2.0_r8
@@ -333,193 +332,6 @@ contains
     deallocate(src_array, mask_src, dst_array, dst_array_t)
 
   end subroutine test_gridmap_areaave_srcmask
-
-  !------------------------------------------------------------------------------
-  subroutine test_gridmap_areaave_srcmask2
-
-    implicit none
-
-    type(gridmap_type) :: gridmap
-    character(len=128) :: testname
-
-    real(r8), allocatable :: src_array(:)
-    real(r8), allocatable :: mask_src(:)
-    real(r8), allocatable :: dst_array(:)
-    real(r8), allocatable :: mask_dst(:)
-    real(r8), allocatable :: dst_array_t(:)
-
-    real(r8), parameter :: mask_dst_min = 0.0_r8
-    real(r8), parameter :: nodata = -1._r8
-    real(r8), parameter :: eps = 1.e-13_r8
-
-    character(len=*), parameter :: subname = 'test_gridmap_areaave_srcmask2'
-
-    ! Note about the gridmaps for the tests here:
-    ! For most tests here, the test arrays are: (1) simple case, (2) the main case to
-    ! test, (3) simple case. Thus, the main case in question is #2 of 3, and we're always
-    ! basically just testing one scenario in each call to the subroutine (rather than
-    ! doing a bunch of tests at once, which could make setting up the test arrays more
-    ! error-prone).
-
-    ! Set up a gridmap with 0 weight of overlap on dest #2
-    gridmap%na = 4
-    gridmap%nb = 3
-    gridmap%ns = 4
-    allocate(gridmap%src_indx(gridmap%ns), &
-             gridmap%dst_indx(gridmap%ns), &
-             gridmap%wovr    (gridmap%ns), &
-             gridmap%frac_dst(gridmap%nb))
-    gridmap%src_indx = (/1,2,3,4/)
-    gridmap%dst_indx = (/1,1,3,3/)
-    gridmap%wovr     = (/0.75_r8,0.25_r8, &  ! weights of sources 1:2 on dest 1
-                         0.25_r8,0.75_r8/)   ! weights of sources 3:4 on test 3
-    gridmap%frac_dst = (/1.0, 0.0, 1.0/)
-    gridmap%set = 'gridmap_IsSet'
-    allocate(src_array  (gridmap%na), &
-         mask_src   (gridmap%na), &
-         dst_array  (gridmap%nb), &
-         mask_dst   (gridmap%nb), &
-         dst_array_t(gridmap%nb))
-    testname = 'no overlap'
-    src_array = (/0.1_r8,0.2_r8,0.3_r8,0.4_r8/)
-    mask_src(:) = 1.0_r8
-    mask_dst(:) = 1.0_r8
-    dst_array_t = (/0.125_r8, nodata, 0.375_r8/)
-    call gridmap_areaave(gridmap, src_array, dst_array, nodata, mask_src, mask_dst, mask_dst_min)
-    call test_close(dst_array, dst_array_t, eps, modname//' -- '//subname//' -- '//trim(testname))
-    deallocate(gridmap%src_indx, gridmap%dst_indx, gridmap%wovr, gridmap%frac_dst)
-    deallocate(src_array, mask_src, dst_array, mask_dst, dst_array_t)
-
-    ! Set up a gridmap with a single point overlapping dest #2
-    gridmap%na = 5
-    gridmap%nb = 3
-    gridmap%ns = 5
-    allocate(gridmap%src_indx(gridmap%ns), &
-             gridmap%dst_indx(gridmap%ns), &
-             gridmap%wovr    (gridmap%ns), &
-             gridmap%frac_dst(gridmap%nb))
-    gridmap%src_indx = (/1,2,3,4,5/)
-    gridmap%dst_indx = (/1,1,2,3,3/)
-    gridmap%wovr     = (/0.75_r8,0.25_r8, &  ! weights of sources 1:2 on dest 1
-                         1.0_r8, &           ! weight of source 3 on dest 2
-                         0.25_r8,0.75_r8/)   ! weights of sources 4:5 on test 3
-    gridmap%frac_dst = (/1.0, 1.0, 1.0/)
-    gridmap%set = 'gridmap_IsSet'
-    allocate(src_array  (gridmap%na), &
-         mask_src   (gridmap%na), &
-         dst_array  (gridmap%nb), &
-         mask_dst   (gridmap%nb), &
-         dst_array_t(gridmap%nb))
-    testname = 'single overlap'
-    src_array = (/0.1_r8,0.2_r8,0.5_r8,0.3_r8,0.4_r8/)
-    mask_src(:) = 1.0_r8
-    mask_dst(:) = 1.0_r8
-    dst_array_t = (/0.125_r8, 0.5_r8, 0.375_r8/)
-    call gridmap_areaave(gridmap, src_array, dst_array, nodata, mask_src, mask_dst, mask_dst_min)
-    call test_close(dst_array, dst_array_t, eps, modname//' -- '//subname//' -- '//trim(testname))
-
-    ! Now change the overlap point to have src_mask=0
-    testname = 'single overlap with 0 src_mask'
-    mask_src(3) = 0.0_r8
-    mask_dst(:) = 1.0_r8
-    dst_array_t(2) = nodata
-    call gridmap_areaave(gridmap, src_array, dst_array, nodata, mask_src, mask_dst, mask_dst_min)
-    call test_close(dst_array, dst_array_t, eps, modname//' -- '//subname//' -- '//trim(testname))
-
-    deallocate(gridmap%src_indx, gridmap%dst_indx, gridmap%wovr, gridmap%frac_dst)
-    deallocate(src_array, mask_src, dst_array, mask_dst, dst_array_t)
-
-    ! Set up a gridmap for the remaining tests
-    ! This gridmap will have 3 src cells, 9 dest cells, and:
-    ! src 1: just overlaps with dst 1
-    ! src 2: overlaps with dst 1 & dst 2
-    ! src 3..7: just overlaps with dst 2
-    ! src 8: overlaps with dst 2 & dst 3
-    ! src 9: just overlaps with dst 3
-    gridmap%na = 9
-    gridmap%nb = 3
-    gridmap%ns = 11
-    allocate(gridmap%src_indx(gridmap%ns), &
-             gridmap%dst_indx(gridmap%ns), &
-             gridmap%wovr    (gridmap%ns), &
-             gridmap%frac_dst(gridmap%nb))
-    gridmap%src_indx = (/1,2,2,3,4,5,6,7,8,8,9/)
-    gridmap%dst_indx = (/1,1,2,2,2,2,2,2,2,3,3/)
-    gridmap%wovr     = (/0.75_r8,0.25_r8, &  ! weights of sources 1:2 on dest 1
-                         0.05_r8,0.05_r8,0.1_r8,0.3_r8,0.2_r8,0.15_r8,0.15_r8, &  ! weights of sources 2:8 on dest 2
-                         0.25_r8,0.75_r8/)   ! weights of sources 8:9 on test 3
-    gridmap%frac_dst = (/1.0_r8, 1.0_r8, 1.0_r8/)
-    gridmap%set = 'gridmap_IsSet'
-    allocate(src_array  (gridmap%na), &
-         mask_src   (gridmap%na), &
-         dst_array  (gridmap%nb), &
-         mask_dst   (gridmap%nb), &
-         dst_array_t(gridmap%nb))
-
-   
-    testname='multiple overlaps, all the same value'
-    src_array = (/0.1_r8, 0.5_r8, 0.5_r8, 0.5_r8, 0.5_r8, 0.5_r8, 0.5_r8, 0.5_r8, 0.6_r8/)
-    mask_src(:) = 1.0_r8
-    mask_dst(:) = 1.0_r8
-    dst_array_t = (/0.2_r8, 0.5_r8, 0.575_r8/)
-    call gridmap_areaave(gridmap, src_array, dst_array, nodata, mask_src, mask_dst, mask_dst_min)
-    call test_close(dst_array, dst_array_t, eps, modname//' -- '//subname//' -- '//trim(testname))
-
-    testname='multiple overlaps, different values'
-    src_array = (/0.1_r8, 0.2_r8, 0.3_r8, 0.7_r8, 0.5_r8, 1.5_r8, 0.5_r8, 1.7_r8, 1.8_r8/)
-    mask_src(:) = 1.0_r8
-    mask_dst(:) = 1.0_r8
-    dst_array_t = (/0.125_r8, 0.875_r8, 1.775_r8/)
-    call gridmap_areaave(gridmap, src_array, dst_array, nodata, mask_src, mask_dst, mask_dst_min)
-    call test_close(dst_array, dst_array_t, eps, modname//' -- '//subname//' -- '//trim(testname))
-
-    testname='multiple overlaps, different values, dst mask'
-    src_array = (/0.1_r8, 0.2_r8, 0.3_r8, 0.7_r8, 0.5_r8, 1.5_r8, 0.5_r8, 1.7_r8, 1.8_r8/)
-    mask_src(:) = 1.0_r8
-    mask_dst(:) = (/1.0_r8, 0.0_r8, 1.0_r8/)
-    dst_array_t = (/0.125_r8, nodata, 1.775_r8/)
-    call gridmap_areaave(gridmap, src_array, dst_array, nodata, mask_src, mask_dst, mask_dst_min)
-    call test_close(dst_array, dst_array_t, eps, modname//' -- '//subname//' -- '//trim(testname))
-
-    testname='multiple overlaps, different values, srcmask'
-    src_array = (/0.1_r8, 0.2_r8, 0.3_r8, 0.7_r8, 0.5_r8, 1.5_r8, 0.5_r8, 1.7_r8, 1.8_r8/)
-    mask_src(:) = (/1.0_r8, 1.0_r8, 0.0_r8, 0.5_r8, 1.0_r8, 0.5_r8, 0.0_r8, 1.0_r8, 1.0_r8/)
-    mask_dst(:) = 1.0_r8
-    dst_array_t = (/0.125_r8, 0.923076923076923_r8, 1.775_r8/)
-    call gridmap_areaave(gridmap, src_array, dst_array, nodata, mask_src, mask_dst, mask_dst_min)
-    call test_close(dst_array, dst_array_t, eps, modname//' -- '//subname//' -- '//trim(testname))
-
-    ! dividing the weights by 2 and dividing mask_src by a constant shouldn't affect the mean
-    testname='weights divided by 2'
-    gridmap%wovr(:) = gridmap%wovr(:) / 2.0_r8
-    gridmap%frac_dst(:) = gridmap%frac_dst(:) / 2.0_r8
-    src_array = (/0.1_r8, 0.2_r8, 0.3_r8, 0.7_r8, 0.5_r8, 1.5_r8, 0.5_r8, 1.7_r8, 1.8_r8/)
-    mask_src(:) = 0.25_r8
-    mask_dst(:) = 1.0_r8
-    dst_array_t = (/0.125_r8, 0.875_r8, 1.775_r8/)
-    call gridmap_areaave(gridmap, src_array, dst_array, nodata, mask_src, mask_dst, mask_dst_min)
-    call test_close(dst_array, dst_array_t, eps, modname//' -- '//subname//' -- '//trim(testname))
-    ! restore wovr & frac_dst
-    gridmap%wovr(:) = gridmap%wovr(:) * 2.0_r8
-    gridmap%frac_dst(:) = gridmap%frac_dst(:) * 2.0_r8
-
-    ! using frac_dst > 1 should be okay
-    testname='frac_dst > 1'
-    gridmap%wovr(:) = gridmap%wovr(:) * 2.0_r8
-    gridmap%frac_dst(:) = gridmap%frac_dst(:) * 2.0_r8
-    src_array = (/0.1_r8, 0.2_r8, 0.3_r8, 0.7_r8, 0.5_r8, 1.5_r8, 0.5_r8, 1.7_r8, 1.8_r8/)
-    mask_src(:) = 0.25_r8
-    mask_dst(:) = 1.0_r8
-    dst_array_t = (/0.125_r8, 0.875_r8, 1.775_r8/)
-    call gridmap_areaave(gridmap, src_array, dst_array, nodata, mask_src, mask_dst, mask_dst_min)
-    call test_close(dst_array, dst_array_t, eps, modname//' -- '//subname//' -- '//trim(testname))
-    ! restore wovr & frac_dst
-    gridmap%wovr(:) = gridmap%wovr(:) / 2.0_r8
-    gridmap%frac_dst(:) = gridmap%frac_dst(:) / 2.0_r8
-    
-    deallocate(src_array, mask_src, dst_array, mask_dst, dst_array_t)
-
-  end subroutine test_gridmap_areaave_srcmask2
 
   !------------------------------------------------------------------------------
   subroutine test_gridmap_areastddev

--- a/tools/mksurfdata_map/unit_testers/test_mkindexmapMod.F90
+++ b/tools/mksurfdata_map/unit_testers/test_mkindexmapMod.F90
@@ -9,7 +9,6 @@ module test_mkindexmapMod
    private
 
    public :: test_get_dominant_indices
-   public :: test_filter_same
    public :: test_lookup_2d
    public :: test_lookup_2d_netcdf
    public :: test_which_max   
@@ -29,6 +28,7 @@ contains
       character(len=128) :: testname
 
       integer, allocatable :: src_array(:)
+      integer, allocatable :: mask_src(:)
       integer, allocatable :: dst_array(:)
       integer, allocatable :: dst_array_t(:)
       logical, allocatable :: filter(:)
@@ -63,18 +63,20 @@ contains
                            0.1,0.1,0.1,0.3,0.2,0.2,0.2, &  ! weights of sources 2:8 on dest 2
                            0.25,0.75/)   ! weights of sources 8:9 on test 3
       allocate(src_array  (gridmap%na), &
+               mask_src   (gridmap%na), &
                dst_array  (gridmap%nb), &
                dst_array_t(gridmap%nb), &
                filter     (gridmap%ns))
 
       testname = 'basic test, all unique'
       src_array = (/1, 2, 3, 4, 5, 6, 7, 8, 9/)
+      mask_src = (/1, 1, 1, 1, 1, 1, 1, 1, 1/)  ! same for all the tests
       minval = 1
       maxval = 9
       nodata = -1
       ! dst 2 takes its value from src 5 because it has the largest weight:
       dst_array_t = (/1, 5, 9/)
-      call get_dominant_indices(gridmap, src_array, dst_array, minval, maxval, nodata)
+      call get_dominant_indices(gridmap, src_array, dst_array, minval, maxval, nodata, mask_src=mask_src)
       call test_is(dst_array, dst_array_t, modname//' -- '//subname//' -- '//trim(testname))
       
       testname = 'basic test, some duplicates'
@@ -83,7 +85,7 @@ contains
       maxval = 4
       nodata = -1
       dst_array_t = (/1, 2, 1/)
-      call get_dominant_indices(gridmap, src_array, dst_array, minval, maxval, nodata)
+      call get_dominant_indices(gridmap, src_array, dst_array, minval, maxval, nodata, mask_src=mask_src)
       call test_is(dst_array, dst_array_t, modname//' -- '//subname//' -- '//trim(testname))
 
       testname = 'minval not 1'
@@ -92,7 +94,7 @@ contains
       maxval = 6
       nodata = -1
       dst_array_t = (/3, 4, 3/)
-      call get_dominant_indices(gridmap, src_array, dst_array, minval, maxval, nodata)
+      call get_dominant_indices(gridmap, src_array, dst_array, minval, maxval, nodata, mask_src=mask_src)
       call test_is(dst_array, dst_array_t, modname//' -- '//subname//' -- '//trim(testname))
 
       testname = 'single non-zero source value'
@@ -101,7 +103,7 @@ contains
       maxval = 2
       nodata = -1
       dst_array_t = (/1, 2, 1/)
-      call get_dominant_indices(gridmap, src_array, dst_array, minval, maxval, nodata)
+      call get_dominant_indices(gridmap, src_array, dst_array, minval, maxval, nodata, mask_src=mask_src)
       call test_is(dst_array, dst_array_t, modname//' -- '//subname//' -- '//trim(testname))
 
       testname = 'single value within given min-max range'
@@ -110,7 +112,7 @@ contains
       maxval = 2
       nodata = -1
       dst_array_t = (/1, 2, 1/)
-      call get_dominant_indices(gridmap, src_array, dst_array, minval, maxval, nodata)
+      call get_dominant_indices(gridmap, src_array, dst_array, minval, maxval, nodata, mask_src=mask_src)
       call test_is(dst_array, dst_array_t, modname//' -- '//subname//' -- '//trim(testname))
       
       testname = 'no valid values'
@@ -119,7 +121,7 @@ contains
       maxval = 2
       nodata = -1
       dst_array_t = (/1, nodata, 1/)
-      call get_dominant_indices(gridmap, src_array, dst_array, minval, maxval, nodata)
+      call get_dominant_indices(gridmap, src_array, dst_array, minval, maxval, nodata, mask_src=mask_src)
       call test_is(dst_array, dst_array_t, modname//' -- '//subname//' -- '//trim(testname))
 
       testname = 'some filters false'
@@ -131,7 +133,7 @@ contains
                  .false., .true., .true., .true., .false., .true., .true., &
                  .true., .true./)
       dst_array_t = (/1, 4, 1/)
-      call get_dominant_indices(gridmap, src_array, dst_array, minval, maxval, nodata, filter=filter)
+      call get_dominant_indices(gridmap, src_array, dst_array, minval, maxval, nodata, filter=filter, mask_src=mask_src)
       call test_is(dst_array, dst_array_t, modname//' -- '//subname//' -- '//trim(testname))
       
       testname = 'all filters false'
@@ -143,7 +145,7 @@ contains
                  .false., .false., .false., .false., .false., .false., .false., &
                  .true., .true./)      
       dst_array_t = (/1, nodata, 1/)
-      call get_dominant_indices(gridmap, src_array, dst_array, minval, maxval, nodata, filter=filter)
+      call get_dominant_indices(gridmap, src_array, dst_array, minval, maxval, nodata, filter=filter, mask_src=mask_src)
       call test_is(dst_array, dst_array_t, modname//' -- '//subname//' -- '//trim(testname))
 
       ! Modify gridmap weights for the following test
@@ -156,7 +158,7 @@ contains
       maxval = 2
       nodata = -1
       dst_array_t = (/1, nodata, 1/)
-      call get_dominant_indices(gridmap, src_array, dst_array, minval, maxval, nodata)
+      call get_dominant_indices(gridmap, src_array, dst_array, minval, maxval, nodata, mask_src=mask_src)
       call test_is(dst_array, dst_array_t, modname//' -- '//subname//' -- '//trim(testname))
       
       ! Make a new gridmap for the following test;
@@ -191,118 +193,13 @@ contains
       maxval = 9
       nodata = -1
       dst_array_t = (/1, 6, 1, nodata, 5/)
-      call get_dominant_indices(gridmap, src_array, dst_array, minval, maxval, nodata)
+      call get_dominant_indices(gridmap, src_array, dst_array, minval, maxval, nodata, mask_src=mask_src)
       call test_is(dst_array, dst_array_t, modname//' -- '//subname//' -- '//trim(testname))
 
       deallocate(gridmap%src_indx, gridmap%dst_indx, gridmap%wovr, & 
                  src_array, dst_array_t, filter)
 
    end subroutine test_get_dominant_indices
-!------------------------------------------------------------------------------
-
-!------------------------------------------------------------------------------
-   subroutine test_filter_same
-      
-      use mkgridmapMod, only : gridmap_type
-
-      implicit none
-
-      type(gridmap_type) :: gridmap
-      character(len=128) :: testname
-      
-      integer, allocatable :: src_array(:)
-      integer, allocatable :: dst_array(:)
-      logical, allocatable :: filter(:)
-      logical, allocatable :: filter_t(:)
-      integer :: nodata
-
-      character(len=*), parameter :: subname = 'test_filter_same'
-      
-      ! Set up a gridmap that will be used for most tests, and allocate corresponding
-      ! arrays:
-      ! Note that, for most tests here, the test arrays are: (1) simple case, (2) the main
-      ! case to test, (3) simple case. Thus, the main case in question is #2 of 3, and
-      ! we're always basically just testing one scenario in each call to the subroutine
-      ! (rather than doing a bunch of tests at once, which could make setting up the test
-      ! arrays more error-prone).
-
-      ! This gridmap will have 3 src cells, 9 dest cells, and:
-      ! src 1: just overlaps with dst 1
-      ! src 2: overlaps with dst 1 & dst 2
-      ! src 3..7: just overlaps with dst 2
-      ! src 8: overlaps with dst 2 & dst 3
-      ! src 9: just overlaps with dst 3
-      ! Note: I'm not setting some things that aren't used in filter_same
-      gridmap%na = 9
-      gridmap%nb = 3
-      gridmap%ns = 11
-      allocate(gridmap%src_indx(gridmap%ns), &
-               gridmap%dst_indx(gridmap%ns))
-      gridmap%src_indx = (/1,2,2,3,4,5,6,7,8,8,9/)
-      gridmap%dst_indx = (/1,1,2,2,2,2,2,2,2,3,3/)
-      allocate(src_array  (gridmap%na), &
-               dst_array  (gridmap%nb), &
-               filter     (gridmap%ns), &
-               filter_t   (gridmap%ns))
-
-      testname = 'maintain false values in filter'
-      src_array(:) = 1
-      dst_array(:) = 1
-      filter(:) = .true.
-      filter(3) = .false.
-      filter(5) = .false.
-      filter_t(:) = .true.
-      filter_t(3) = .false.
-      filter_t(5) = .false.
-      call filter_same(gridmap, filter, src_array, dst_array)
-      call test_is(filter, filter_t, modname//' -- '//subname//' -- '//trim(testname))
-      
-      testname = 'dst_array = nodata in some places'
-      nodata = -1
-      src_array(:) = 1
-      src_array(5) = nodata  ! make sure that even when src_array = dst_array = nodata,
-                             ! we still end up with filter = false
-      dst_array = (/1, nodata, 1/)
-      filter(:) = .true.
-      filter_t(:) = .true.
-      filter_t(3:9) = .false.  ! false for all overlaps with dst #2
-      call filter_same(gridmap, filter, src_array, dst_array, nodata=nodata)
-      call test_is(filter, filter_t, modname//' -- '//subname//' -- '//trim(testname))
-
-      testname = 'src_array not equal to dst_array in some places, no nodata argument'
-      src_array(:) = (/1, 1, 1, 1, 2, 3, 1, 3, 1/)
-      dst_array(:) = (/1, 1, 1/)
-      filter(:) = .true.
-      ! src_array index: 1      2      2      3      4       5       6      7       8       8      9
-      filter_t(:) = (/.true.,.true.,.true.,.true.,.true.,.false.,.false.,.true.,.false.,.false.,.true./)
-      call filter_same(gridmap, filter, src_array, dst_array)
-      call test_is(filter, filter_t, modname//' -- '//subname//' -- '//trim(testname))
-
-      testname = 'src_array not equal to dst_array in some places, nodata never applies'
-      nodata = -1
-      src_array(:) = (/1, 1, 1, 1, 2, 3, 1, 3, 1/)
-      dst_array(:) = (/1, 1, 1/)
-      filter(:) = .true.
-      ! src_array index: 1      2      2      3      4       5       6      7       8       8      9
-      filter_t(:) = (/.true.,.true.,.true.,.true.,.true.,.false.,.false.,.true.,.false.,.false.,.true./)
-      call filter_same(gridmap, filter, src_array, dst_array, nodata=nodata)
-      call test_is(filter, filter_t, modname//' -- '//subname//' -- '//trim(testname))
-      
-      testname = 'combination of false filter, src_array not equal to dst_array, and nodata'
-      nodata = -1
-      src_array(:) = (/1, 2, 1, 2, 1, 2, 1, 2, 1/)
-      dst_array(:) = (/nodata, 1, 1/)
-      filter(:) = .true.
-      filter(4) = .false.
-      filter_t(:) = (/.false.,.false.,.false.,.false.,.false.,.true.,.false.,.true.,.false.,.false.,.true./)
-      call filter_same(gridmap, filter, src_array, dst_array, nodata=nodata)
-      call test_is(filter, filter_t, modname//' -- '//subname//' -- '//trim(testname))
-
-
-      deallocate(gridmap%src_indx, gridmap%dst_indx, & 
-                 src_array, dst_array, filter, filter_t)
-
-   end subroutine test_filter_same
 !------------------------------------------------------------------------------
 
 !------------------------------------------------------------------------------

--- a/tools/mksurfdata_map/unit_testers/test_mkindexmapMod.F90
+++ b/tools/mksurfdata_map/unit_testers/test_mkindexmapMod.F90
@@ -70,7 +70,7 @@ contains
 
       testname = 'basic test, all unique'
       src_array = (/1, 2, 3, 4, 5, 6, 7, 8, 9/)
-      mask_src = (/1, 1, 1, 1, 1, 1, 1, 1, 1/)  ! same for all the tests
+      mask_src(:) = 1  ! same for all the tests
       minval = 1
       maxval = 9
       nodata = -1

--- a/tools/mksurfdata_map/unit_testers/test_mksurfdata_map.F90
+++ b/tools/mksurfdata_map/unit_testers/test_mksurfdata_map.F90
@@ -20,14 +20,12 @@ program mksurfdata_map_unit_tester
    call test_slightly_above
 
    ! Test mkgridmapMod
-   call test_gridmap_areaave_default
+   call test_gridmap_areaave_no_srcmask
    call test_gridmap_areaave_srcmask
-   call test_gridmap_areaave_srcmask2
    call test_gridmap_areastddev
 
    ! Test mkindexmapMod
    call test_get_dominant_indices
-   call test_filter_same
    call test_lookup_2d
    call test_lookup_2d_netcdf
    call test_which_max

